python/cugraph/traversal/bfs_wrapper.pyx

```diff
@@ -17,72 +17,88 @@
     if not 0 <= start < num_verts:
         raise ValueError("Starting vertex should be between 0 to number of vertices")
 
+    # Step 6: Generate the cudf.DataFrame result
+    #         Current implementation expects int (signed 32-bit) for distance
     df = cudf.DataFrame()
     df['vertex'] = cudf.Series(np.zeros(num_verts, dtype=np.int32))
     df['distance'] = cudf.Series(np.zeros(num_verts, dtype=np.int32))
     df['predecessor'] = cudf.Series(np.zeros(num_verts, dtype=np.int32))
-    cdef gdf_column c_vertex_col = get_gdf_column_view(df['vertex'])
-    #cdef uintptr_t c_distance_ptr = get_column_data_ptr(df['distance']._column)
-    cdef uintptr_t c_distance_ptr = df['distance'].__cuda_array_interface__['data'][0]
-    #cdef uintptr_t c_predecessors_ptr = get_column_data_ptr(df['predecessor']._column)
-    cdef uintptr_t c_predecessors_ptr = df['predecessor'].__cuda_array_interface__['data'][0]
-
-    g.adjList.get_vertex_identifiers(&c_vertex_col)
 
-    c_bfs.bfs[int](g, <int*>c_distance_ptr, <int*>c_predecessors_ptr, <int>start)
+    # Step 7: Associate <uintptr_t> to cudf Series
+    c_identifier_ptr = df['vertex'].__cuda_array_interface__['data'][0]
+    c_distance_ptr = df['distance'].__cuda_array_interface__['data'][0]
+    c_predecessor_ptr = df['predecessor'].__cuda_array_interface__['data'][0]
 
+    # Step 8: Proceed to BFS
+    # TODO: [int, int, float] or may add an explicit [int, int, int] in graph.cu?
+    graph_float = GraphCSR[int, int, float](<int*> c_offsets_ptr,
+                                            <int*> c_indices_ptr,
+                                            <float*> NULL,
+                                            num_verts,
+                                            num_edges)
+    graph_float.get_vertex_identifiers(<int*> c_identifier_ptr)
+    c_bfs.bfs[int, int, float](graph_float,
+                               <int*> c_distance_ptr,
+                               <int*> c_predecessor_ptr,
+                               <int> start,
+                               directed)
+    # Step 9: Renumber before return
+    #         It is only required to renumber vertex and predecessors
     if input_graph.renumbered:
-        if isinstance(input_graph.edgelist.renumber_map, cudf.DataFrame):
-            n_cols = len(input_graph.edgelist.renumber_map.columns) - 1
-            unrenumbered_df_ = df.merge(input_graph.edgelist.renumber_map, left_on='vertex', right_on='id', how='left').drop(['id', 'vertex'])
-            unrenumbered_df = unrenumbered_df_.merge(input_graph.edgelist.renumber_map, left_on='predecessor', right_on='id', how='left').drop(['id', 'predecessor'])
-            unrenumbered_df.columns = ['distance']+['vertex_'+str(i) for i in range(n_cols)]+['predecessor_'+str(i) for i in range(n_cols)]
-            cols = unrenumbered_df.columns.to_list()
-            df = unrenumbered_df[cols[1:n_cols+1] + [cols[0]] + cols[n_cols:]]
```

Comment:
Is this still supposed to handled?

Reply:
yes, that looks like the unrenumbering step

Reply:
The lines 77-88 are for multicolumn and single column unrenumbering. Better to leave it as is. It is subject to changes in further releases as we change our renumbering/unrenumbering strategy.

Reply:
Reverted it in https://github.com/rapidsai/cugraph/pull/764/commits/1528aebe9ea39d6ac0060b1068a76f8c2a36056c, and added a FIXME about it